### PR TITLE
Document the new network plan

### LINKED ISSUE #24: Document new network plan

### PR DESCRIPTION
**Pull request by nbraud**
Closes #24

### CODE REVIEW

**Comment by nbraud** [comment]
@equinox0815 I'm sorry I didn't manage to finish before leaving.

**Comment by nbraud** [comment]
@equinox0815 BTW, I just realised that keeping `89.106.211.32/27` on the members LAN (or, in a first time, the guests' LAN) and moving services to `89.106.211.64/27` (which we can do one by one) would let us do a more incremental move; what do you think?

**Comment by equinox0815** [comment]
Yes sure. I had the same thought already.

**Comment by nbraud** [comment]
@equinox0815 OK, I did that change in the documentation; if everything is OK from your side, I think we can merge it.

File: doc/Network.mdwn [hunk 1]
```diff
@@ -1,29 +1,57 @@
...
-- 255 (`0xFF`) is our Funkfeuer VLAN.
+NOC operates a number of networks, available as tagged VLANs on the core
+switches (one in each half of the hackerspace).  These networks are:
 
-The switches have hostnames `sw{0,1}.mgmt.realraum.at`, and the WiFi access
-points are similarly `ap{0,1}.mgmt.realraum.at`. `0` denotes the main room, and
-`1` denotes Wöhnung 2.
+| Network    | VLAN id | Extra subnets    |
+|------------|---------|------------------|
+| Management |      23 | --               |
```
Comment (equinox0815) [inline doc/Network.mdwn]: Management VLAN is 32 and not 23

File: doc/Network.mdwn [hunk 2]
```diff
@@ -1,29 +1,57 @@
...
+| IoT        |      33 | --               |
+| Services   |      34 | --               |
```
Comment (equinox0815) [inline doc/Network.mdwn]: Net Range 192.168.34.0/24 is currently used for connections on http bridge on Alfred... Not sure if we want to make this the service VLAN....

File: doc/Network.mdwn [hunk 3]
```diff
@@ -1,29 +1,57 @@
...
+### Conventions
+
+We use a number of conventions to make things more consistent:
+
+- The DNS zone for a given network is `NET.realraum.at`, with the exception
+  of the public services network (which uses `realraum.at`) and of the Funkfeuer
+  VLAN (which has no `realraum.at` zone).
+- Networks using RFC 1918 IP space use the 192.168.VID.0/24 subnet;
+- The gateway for a network is on the last IP for the subnet.
+
```
Comment (equinox0815) [inline doc/Network.mdwn]: Please add an example here...

File: doc/Network.mdwn [hunk 4]
```diff
@@ -1,29 +1,57 @@
...
 
-Moreover, realfunk has an Archer C7 acting as a VLAN-capable switch
-and as a Funkfeuer node; it does **not** provide the realraum SSIDs.
-
 
 ## Physical locations
 
+The switches have hostnames `sw{0,1}.mgmt.realraum.at`, and the WiFi access
+points are similarly `ap{0,1}.mgmt.realraum.at`. `0` denotes the main room, and
+`1` denotes Wöhnung 2.
```
Comment (equinox0815) [inline doc/Network.mdwn]: LOL wöhnung!

File: doc/Network/overview.dot [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  node [ shape=none fontname="Liberation Sans" fontcolor="#c0c0c0" ];
+  edge [fontname="Liberation Sans"
+        fontcolor="#c0c0c0" color="#c0c0c0"
+  ];
+
+  public  [ label="Public (36)\n89.106.211.64/27" ]
+  inet    [ label="Internet" ]
+  guests  [ label="Guests (127)" ]
+  members [ label="Members (128)\n89.106.211.32/27" ]
+  mgmt    [ label="Management (23)" ]
```
Comment (equinox0815) [inline doc/Network/overview.dot]: again: 23 -> 32
at least this is the case right now... we could change it of course... Is this a proposal to change it or is it a typo?

Comment (nbraud) [reply doc/Network/overview.dot]: Typo (or rather, I wrote it the wrong way around when I documented it on paper)

File: doc/Network.mdwn (hunk 2 above)
Comment (nbraud) [reply doc/Network.mdwn]: IIRC, that was the proposal when we discussed (and what I wrote on paper).

To clarify (I just realised this needs some extra explanatory text):
- services reachable from outside (so, Nginx & friends) live in the “public” VLAN;
- the “services” VLAN is for not-frontend services, that aren't reachable from anywhere else (unless they explicitly have an extra interface).

File: doc/Network.mdwn (hunk 4 above)
Comment (nbraud) [reply doc/Network.mdwn]: I probably wrote that quite late  >_<'
Not too sure what to put, though.

**Comment by nbraud** [comment]
@equinox0815 In case you missed it, I handled your previous comments, except the [service VLAN thing](https://github.com/realraum/noc/pull/25#discussion_r199958117) which I clarified.

File: doc/Network.mdwn (hunk 2 above)
Comment (nbraud) [reply doc/Network.mdwn]: Would naming it “backend” or somesuch be clearer?

File: doc/Network.mdwn [hunk 6]
```diff
@@ -1,37 +1,67 @@
...
+| Guests     |     127 | --               |
+| Members    |     128 | 89.106.211.32/27 |
+| `0xFF`     |     255 | --               |
+
+
+### Conventions
+
+We use a number of conventions to make things more consistent:
+
+- The DNS zone for a given network is `NET.realraum.at`, with the exception
```
Comment (equinox0815) [inline doc/Network.mdwn]: hmmm NET is not really the name as seen above i.e. Managment vlan has hostname `mgmt.realraum.at`
I propose using the "short" name above and add a description column to it so the above table looks like this:

```
+| Name | VLAN id | Extra subnets    | Description                        |
+|------|---------|------------------|------------------------------------|
+| mgmt |    32   | --               | Managment interfaces for routers, switches, etc... |
...
```
The description field will also help to understand the difference between `svc` and `public`.

Comment (nbraud) [reply doc/Network.mdwn]: :+1: makes sense.

Comment (nbraud) [reply doc/Network.mdwn]: Done  :)

**Pull request merged**